### PR TITLE
Update EIP-7702: 7702 validity

### PR DESCRIPTION
**Pull request by lightclient**
Repost from #8840 as it should not have automatically merged.

> The past few weeks we've been discussing validity conditions for 7702 transactions. Originally @rakita added the decoding limits for auth list items in #8746. Now we're looking at further constraining validity in #8833 to disallow chain ids other than zero or the current chain's id.

> My philosophy here is 1) tx validity should be as simple as possible to ascertain and 2) we should retain as much similarity across tx types as possible.

> The motivation for 1) is that tx originators are incentivized to not send junk authorizations with their txs, therefore they won't. What's the point of adding more constraints into the protocol when they aren't necessary?

> For 2), I believe it's extremely important to follow existing conventions when possible. If every single tx type is designed as an ornate protocol fixture by core devs of that era, we will have a mess of formats that are extremely difficult to reason about as a whole or onboard new developers to understanding. Of course this doesn't mean we shouldn't improve the status quo when possible (see: separating chain id back out from `v` value), but unless the benefit is significant, let's not deviate from other tx formats.

### CODE REVIEW

**Comment by rakita** [comment]
My stance on this is:
* Sanity limits need to be made. We i think all agree on this.
* @jochem-brouwer asked to add checks on `s-value` and `v`. I don't have a stance on this.
   * But **If** `s-value` and `v` are checked, I would request that we add the check for `chain_Id`
 
 
Side question for this PR, why not use u8 for `v`?

**Comment by lightclient** [comment]
Agree sanity limits need to exist, I just think we should keep them minimal and inline with existing types.

> But If s-value and v are checked, I would request that we add the check for chain_Id

All of these things should be checked, they just shouldn't affect the transactions validity.

> why not use u8 for v?

We bound `v < 2**256-1` everywhere else, it seems weird to do something different here.

File: EIPS/eip-7702.md [hunk 1]
```diff
@@ -44,19 +44,22 @@ rlp([chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, dest
...
 
 The `authorization_list` is a list of tuples that store the address to code which the signer desires to execute in the context of their EOA. The transaction is considered invalid if the length of `authorization_list` is zero.
 
+The transaction is also considered invalid when any field in an authorization
+tuple cannot fit within the following bounds:
+
+```python
+assert auth.chain_id < 2**256
+assert auth.nonce < 2**64
+assert len(auth.address) == 20
```
Comment (drortirosh) [inline EIPS/eip-7702.md:57]: shouldn't it be `len(auth.address)<=20` ?

Comment (lightclient) [reply EIPS/eip-7702.md:57]: no, address is a fixed length item.

File: EIPS/eip-7702.md [hunk 2]
```diff
@@ -44,19 +44,22 @@ rlp([chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, dest
...
+assert auth.y_parity < 2**256
+assert auth.r < 2**256
+assert auth.z < 2**256
```
Comment (marioevz) [inline EIPS/eip-7702.md:60]:
```suggestion
assert auth.s < 2**256
```

**Comment by lightclient** [comment]
Seems like there was good support for this in ACDE 196. Still need to look more into constraining the `v` value, but this PR can be merged in the meantime.

File: EIPS/eip-7702.md [hunk 3]
```diff
@@ -44,19 +44,22 @@ rlp([chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, dest
...
 The fields `chain_id`, `nonce`, `max_priority_fee_per_gas`, `max_fee_per_gas`, `gas_limit`, `destination`, `value`, `data`, and `access_list` of the outer transaction follow the same semantics as [EIP-4844](./eip-4844.md). *Note, this means a null destination is not valid.*
 
 The `authorization_list` is a list of tuples that store the address to code which the signer desires to execute in the context of their EOA. The transaction is considered invalid if the length of `authorization_list` is zero.
 
+The transaction is also considered invalid when any field in an authorization
+tuple cannot fit within the following bounds:
+
+```python
+assert auth.chain_id < 2**256
+assert auth.nonce < 2**64
```
Comment (jochem-brouwer) [inline EIPS/eip-7702.md:56]: Have a slight nit about this. (Sorry, just noticed after-merge). By https://eips.ethereum.org/EIPS/eip-2681 should this not be `2**64 - 1`? Or is this checkout outside tx validity, but instead inside EVM?

Comment (chfast) [reply EIPS/eip-7702.md:56]: It is, but specific to the tx sender. The idea here is to only give numeric boundaries here and leave interpretation to other subsystems.

Comment (gumb0) [reply EIPS/eip-7702.md:56]: > It is, but specific to the tx sender.

The nonce of authority is also incremented during delegation setting, so it is important to check `nonce < 2**64 - 1` here.

I've made this change in https://github.com/ethereum/EIPs/pull/8905

**Pull request merged**